bot/exts/pride/pride_leader.py

```diff
@@ -0,0 +1,115 @@
+        """
+        Generates Invalid Embed.
+
+        The invalid embed contains a list of closely matched names of the invalid pride
+        leader the user gave. If no closely matched names are found it would list all
+        the available pride leader names.
+
+        Wikipedia is a useful place to learn about pride leaders and we don't have all
+        the pride leaders, so the bot would add a field containing the wikipedia
+        command to execute.
+        """
+        embed = discord.Embed(
+            color=constants.Colours.soft_red
+        )
+        valid_names = []
+        pride_leader = pride_leader.title()
+        for name in PRIDE_RESOURCE:
+            if fuzz.ratio(pride_leader, name) >= MINIMUM_FUZZ_RATIO:
+                valid_names.append(name)
+
+        if not valid_names:
+            valid_names = ", ".join(PRIDE_RESOURCE)
+            error_msg = "Sorry your input didn't match any stored names, here is a list of available names:"
+        else:
+            valid_names = "\n".join(valid_names)
+            error_msg = "Did you mean?"
+
+        embed.description = f"{error_msg}\n```{valid_names}```"
+        embed.set_footer(text="To add more pride leaders, feel free to open a pull request!")
+
+        return embed
+
+    def embed_builder(self, leader_name: str) -> discord.Embed:
+        """Generate an Embed with information about a pride leader."""
+        embed = discord.Embed(
+            title=leader_name,
+            description=PRIDE_RESOURCE[leader_name]["About"],
+            color=constants.Colours.blue
+        )
+        embed.add_field(
+            name="Known for",
+            value=PRIDE_RESOURCE[leader_name]["Known for"],
+            inline=False
+        )
+        embed.add_field(
+            name="D.O.B and Birth place",
+            value=PRIDE_RESOURCE[leader_name]["Born"],
+            inline=False
+        )
+        embed.add_field(
+            name="Awards and honors",
+            value=PRIDE_RESOURCE[leader_name]["Awards"],
+            inline=False
+        )
+        embed.add_field(
+            name="For More Information",
+            value=f"Do `{constants.Client.prefix}wiki {leader_name}`"
+                  f" in <#{constants.Channels.community_bot_commands}>",
+            inline=False
+        )
+        embed.set_thumbnail(url=PRIDE_RESOURCE[leader_name]["url"])
+        return embed
+
+    @commands.command(aliases=("pl", "prideleader"))
+    async def pride_leader(self, ctx: commands.Context, *, pride_leader_name: Optional[str]) -> None:
+        """
+        Information about a Pride Leader.
+
+        Returns information about the specified pride leader
+        and if there is no pride leader given, return a random pride leader.
+        """
+        if not pride_leader_name:
+            pride_leader_name = random.choice(list(PRIDE_RESOURCE))
+        else:
+            leader = PRIDE_RESOURCE.get(pride_leader_name.title())
```

Comment:
If you normalize the leader name here, `embed_builder` still gets the unnormalized leader name. Example: `.pl tim cook` raises a `KeyError`.